### ISSUE TITLE
[Discussion] Why gdscript and not something else?

### ISSUE THREAD

**Issue by filipinyo** [AlexeyBond/godot-constraint-solving#7]
Wouldn't the speed be better using C# or something else? I'm not trying to offend anyone, but I'd really like to know what was the benefit of picking gdscript. Read multiple times that iteration is faster in other languages etc.

**Comment by AlexeyBond** [comment]
Sure, it would work much faster in almost any other language.

Not sure, how to answer this question. I guess, I can start by describing how I use Godot - it's mostly on game jams (specifically, [Ludum Dare](https://ldjam.com/users/a-bond/games)) and small experimenal projects. So, I didn't spend time learning and setting up everything necessary to use any better language. Also, I don't know C# and don't have plans to learn it yet. This addon was started as one of small experimental projects - as a demo for a lecture about WFC algorithm. But it has grown too complex too quickly, so I had to make a simpler demo.

To summarize:

Why not C#?
- I don't know it, and don't want to learn it yet
- It doesn't work for web builds at least with current version of Godot. GDScript works, just fine, even with multithreading.

Why not *any other language* using GDExtensions:
- Difficult to setup both for me and users of the addon

Maybe, later I will rewrite this addon, or, at least, it's most critical parts in different language. Most likely, it will be Rust. But for now it will stay with GDScript despite how much it (GDScript) sucks (and not just in performance).
